File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        
        int lewy = 0;
        int prawy = heights.size() - 1;
        
        int maksymalne_pole = 0; 

        while (lewy < prawy) {
            
            int szerokosc = prawy - lewy;
            
            
            int wysokosc_wody = min(heights[lewy], heights[prawy]);
            
            
            int aktualne_pole = szerokosc * wysokosc_wody;
            
            maksymalne_pole = max(maksymalne_pole, aktualne_pole);

            if (heights[lewy] < heights[prawy]) {
                lewy++; 
            } else {
                prawy--; 
            }
        }
        
        return maksymalne_pole;
    }
};

File: Data Structures & Algorithms/is-palindrome/submission-2.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int lewy = 0;
        int prawy = s.Length - 1;

        while(lewy < prawy){
            while(lewy < prawy && !char.IsLetterOrDigit(s[lewy])){
                lewy++;
            }
            while(lewy < prawy && !char.IsLetterOrDigit(s[prawy])){
                prawy--;
            }
            if(char.ToLower(s[lewy]) != char.ToLower(s[prawy])){
                return false;
            }
            lewy++;
            prawy--;
        }
        return true;
    }
}
